opensearch/src/main/java/org/opensearch/sql/opensearch/data/type/OpenSearchTextType.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.opensearch.data.type;
+
+import static org.opensearch.sql.data.type.ExprCoreType.STRING;
+import static org.opensearch.sql.data.type.ExprCoreType.UNKNOWN;
+
+import java.util.List;
+import lombok.EqualsAndHashCode;
+import org.opensearch.sql.data.type.ExprType;
+
+@EqualsAndHashCode(callSuper = false)
```

Comment:
Can you add a javadoc with a link to the opensearch doc?
We should maybe also indicate the differences between `text` and `keyword` types

Reply:
Added in ae11c22.